### ISSUE TITLE
Transactions invalidate nix gc roots

### ISSUE THREAD

**Issue by ysndr** [flox/flox#469]
The env-link created during install will not be recorded by nix as a valid gcroot!
once the transaction is done, we should still set a gc root such that the environment is not garbage collected by nix.

Granted, we don't perform gc's with nix at the moment.

_Originally posted by @ysndr in https://github.com/flox/flox/pull/459#discussion_r1401892751_

**Comment by mkenigs** [comment]
Duplicate of https://github.com/flox/flox/issues/231
